addons/mail/static/src/models/thread/thread.js

```diff
@@ -1363,6 +1363,46 @@ function factory(dependencies) {
             return [['replace', this.activities.filter(activity => activity.state === 'overdue')]];
         }
 
+        /**
+         * Current partner seen infos.
+         *
+         * @private
+         * @returns {[mail.partner_seen_infos]}
+         */
+        _computeCurrentPartnerSeenInfos() {
+            if (!this.messaging) {
+                return [['unlink-all']];
+            }
+            if (!this.messaging.currentPartner) {
+                return [['unlink-all']];
+            }
+            return [['replace', this.partnerSeenInfos.filter(partnerSeenInfo => partnerSeenInfo.partner === this.messaging.currentPartner)]]
```

Comment:
This is a x2one, need to be careful with `replace` with potentially empty value (which won't work because `replace` is equivalent to `link`). Should do `link`/`unlink` appropriately.

(I'm not sure if it's possible to have no partner seen info for current partner but better be safe here, not rely on flux)